lib/cancan/model_adapters/active_record_5_adapter.rb

```diff
@@ -22,9 +22,14 @@ def self.matches_condition?(subject, name, value)
       private
 
       def build_relation(*where_conditions)
-        relation = @model_class.where(*where_conditions)
-        relation = relation.left_joins(joins).distinct if joins.present?
-        relation
+        if joins.present?
+          inner = @model_class.unscoped do
+            @model_class.left_joins(joins).where(*where_conditions)
+          end
+          @model_class.where(id: inner)
```

Comment:
I believe, using `@model_class.primary_key` instead of `:id` would be more safe (as primary key column could be redefined for ActiveRecord model).

```ruby
@model_class.where(@model_class.primary_key => inner)
```